### PR TITLE
operator: Prepare community release v0.4.0

### PR DESCRIPTION
**Pull request by periklis**
**What this PR does / why we need it**:
Prepare Community Loki Operator v0.4.0 release.

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: operator/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## Main
 
+## 0.4.0 (2023-07-25)
```
Comment (xperimental) [inline operator/CHANGELOG.md:3]: Might want to update the date when you plan to release tomorrow.

File: operator/Makefile [hunk 2]
```diff
@@ -21,7 +21,7 @@ LOKI_OPERATOR_NS ?= kubernetes-operators
 # To re-generate a bundle for another specific version without changing the standard setup, you can:
 # - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
-VERSION ?= v0.3.0
+VERSION ?= v0.4.0
```
Comment (xperimental) [inline operator/Makefile:24]: I can only see `$VERSION` used together with `subst` in this file... do we use it anywhere else that I'm missing? If not, we could just remove the `v` and get rid of the `subst`.

File: operator/bundle/openshift/manifests/loki-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -1639,7 +1639,7 @@ spec:
                   value: quay.io/observatorium/api:latest
                 - name: RELATED_IMAGE_OPA
                   value: quay.io/observatorium/opa-openshift:latest
-                image: quay.io/openshift-logging/loki-operator:v0.1.0
+                image: quay.io/openshift-logging/loki-operator:0.1.0
```
Comment (xperimental) [inline operator/bundle/openshift/manifests/loki-operator.clusterserviceversion.yaml:1642]: Was this change intentional?

File: operator/config/manager/kustomization.yaml [hunk 4]
```diff
@@ -6,4 +6,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: quay.io/openshift-logging/loki-operator
-  newTag: v0.1.0
+  newTag: 0.1.0
```
Comment (xperimental) [inline operator/config/manager/kustomization.yaml:9]: Was this change intentional?

**Review by xperimental — COMMENTED** [review]
Looks fine. The comments are mostly optional. The change in the `openshift` bundle will need a change in our build config (assuming it's intentional).

File: operator/config/manager/kustomization.yaml (hunk 4 above)
Comment (periklis) [reply operator/config/manager/kustomization.yaml]: Good catch

**Review by JoaoBraveCoding — APPROVED** [review]
lgtm

**Review by xperimental — APPROVED** [review]
